packages/xsnap/src/xsnap.js

```diff
@@ -96,7 +102,65 @@ export async function xsnap(options) {
   }
 
   /** @type {(opts: import('tmp').TmpNameOptions) => Promise<string>} */
-  const ptmpName = promisify(fs.tmpName);
+  const ptmpName = fs.tmpName && promisify(fs.tmpName);
+
+  const makeLoadSnapshotHandlerWithFS = async () => {
+    assert(snapshotStream);
+    const snapPath = await ptmpName({
+      template: `load-snapshot-${safeHintFromDescription(
+        snapshotDescription,
+      )}-XXXXXX.xss`,
+    });
+
+    const cleanup = async () => fs.unlink(snapPath);
```

Comment:
Will `cleanup` be called temporally later than `afterSpawn`? I'd define them in the same order, so swap these two lines.